controller/src/large_access.rs

```diff
@@ -180,4 +191,80 @@ mod tests {
             "All reads need to sum to the full expected size",
         );
     }
+
+    #[test]
+    fn test_build_large_memory_access_at_boundary() {
+        // test at 256
+        test_build_large_memory_access_impl::<mgmt::MemoryRead, mgmt::cmis::Page>(
+            mgmt::cmis::Page::Upper(UpperPage::new_unbanked(0).unwrap()),
+            192,
+            64,
+        );
+        test_build_large_memory_access_impl::<mgmt::MemoryWrite, mgmt::cmis::Page>(
+            mgmt::cmis::Page::Upper(UpperPage::new_unbanked(0).unwrap()),
+            192,
+            64,
+        );
+        test_build_large_memory_access_impl::<mgmt::MemoryRead, mgmt::sff8636::Page>(
+            mgmt::sff8636::Page::Upper(mgmt::sff8636::UpperPage::new(0).unwrap()),
+            192,
+            64,
+        );
+        test_build_large_memory_access_impl::<mgmt::MemoryWrite, mgmt::sff8636::Page>(
+            mgmt::sff8636::Page::Upper(mgmt::sff8636::UpperPage::new(0).unwrap()),
+            192,
+            64,
+        );
+    }
+
+    #[test]
+    fn test_build_large_memory_access_exceeding_boundary() {
+        // test at 256
+        test_build_large_memory_access_impl::<mgmt::MemoryRead, mgmt::cmis::Page>(
+            mgmt::cmis::Page::Upper(UpperPage::new_unbanked(0).unwrap()),
+            192,
+            65,
+        );
+        test_build_large_memory_access_impl::<mgmt::MemoryWrite, mgmt::cmis::Page>(
+            mgmt::cmis::Page::Upper(UpperPage::new_unbanked(0).unwrap()),
+            192,
+            65,
+        );
+        test_build_large_memory_access_impl::<mgmt::MemoryRead, mgmt::sff8636::Page>(
+            mgmt::sff8636::Page::Upper(mgmt::sff8636::UpperPage::new(0).unwrap()),
+            192,
+            65,
+        );
+        test_build_large_memory_access_impl::<mgmt::MemoryWrite, mgmt::sff8636::Page>(
+            mgmt::sff8636::Page::Upper(mgmt::sff8636::UpperPage::new(0).unwrap()),
+            192,
+            65,
+        );
+    }
+
+    fn test_build_large_memory_access_impl<T, P>(page: P, offset: u8, len: u8)
+    where
+        T: LargeMemoryAccess<P>,
+        P: MemoryPage,
+        mgmt::Page: From<P>,
+        mgmt::MemoryRead: LargeMemoryAccess<P>,
+    {
+        let idx = offset as usize + len as usize;
+        if idx <= 256 {
+            let reads = mgmt::MemoryRead::build_many(page, offset, len)
```

Comment:
I think this should be `T::build_many()`, right? Otherwise, we're always constructing reads, even when `T` is `MemoryWrite`.

Reply:
Oh, this is copied from existing code! Then I think just fix it here, and make an issue to fix the rest of the code.

Reply:
Doing this is much more annoying, since the methods we're calling inside aren't part of the trait bound. I'd ignore this for now, and I'll deal with it in a follow-up issue.